virttest/graphical_console.py

```diff
@@ -229,6 +415,95 @@ def _key_event(self, down, keys):
             events.append(event)
         self._vm.qmp_monitors[0].input_send_event(events)
 
+    def _btn_event(self, event, btn):
+        """
+        Send mouse button event.
+
+        :param event: Button event, defined as True=down, False=up.
+        :param btn: Mouse button.
+        """
+
+        events = [{"type": "btn",
+                   "data": {"down": event,
+                            "button": btn}}]
+        self._vm.monitor.input_send_event(events)
+
+    def _scroll_event(self, event):
+        """
+        Send mouse scroll event.
+
+        :param event: Event as vertical up/down or horizontal left/right.
+                      But qmp only support scroll vertical up/down.
+                      Does not support scroll horizontal left/right currently.
+        """
+
+        if event == "up":
+            self._btn_event(self.BTN_DOWN, self.SCROLL_FORWARD)
+        elif event == "down":
+            self._btn_event(self.BTN_DOWN, self.SCROLL_BACKWARD)
+        else:
+            raise ConsoleNotSupportedError("qmp console doesn't support"
```

Comment:
Seems it won't happen so you can remove the `else` branch.

Reply:
Considering may be need support scroll_left or scroll_right event except scroll_up/down in the future, so add this else branch here, qmp console doesn't support them now.